docs/experimental/closure_glyph_segmentation_complex_conditions.md

```diff
@@ -0,0 +1,140 @@
+
+1. Start with a set of all segments except those to be excluded, called `to_test`.
+2. Initialize a second set of segments, `required`, to the empty set.
+3. Remove a segment `s` from `to_test` and compute the glyph closure of `to_test U required`.
+4. If `glyph` is not found in the closure then add `s` to `required`.
+5. If `to_test` is empty, then return the  sub condition `required`.
+6. Otherwise, go back to step 3.
+
+### Finding the Complete Condition
+
+This section describes the algorithm which finds the complete condition, it utilizes `Finding a Sub Condition`.
+
+Inputs:
+
+* `glyph` to analyze.
+
+Algorithm:
+
+1. Initialize a set of segments `condition` to the empty set.
+2. Execute the `Finding a Sub Condition` algorithm with `condition` as the excluded set.
+3. Union the returned set into `condition`.
+4. Compute the glyph closure of all segments except those in `condition`.
+5. If `glyph` is found in the closure, then more conditions still exist. Go back to step 2.
+6. Return the complete condition, `condition`.
+
+### Initial Font
+
+Any time a closure operation is executed by the above two algorithms it's necessary to union the subset definition 
+for the initial font into the closure input. That's because the closure of the initial font affects what's reachable
+by the segments.
+
+### Why this works
+
+* Any segments which are not part of the true condition will not impact the glyph's presence in the closure (assertion
+  (4)).  As a result they will never be moved into the `required` set and will not be returned by `Finding a Sub
+  Condition`.  Thus any segments returned by `Finding a Sub Condition` are part of the true condition.
+  
+* Each iteration of `Finding a Sub Condition` is guaranteed to select at least one segment since we know that the
+  initial closure always starts with the glyph in it, and the closure of no segments will not have the glyph in it.  So
+  at some point during the algorithm the glyph must be found to not be present. In the first iteration this is a result
+  of assertion (3). For subsequent iterations this is guaranteed by the "additional conditions" check prior to starting
+  the iteration.
+  
+* Since all returned segments from `Finding a Sub Condition` are excluded from future calls, there will be a finite
+  number of `Finding a Sub Condition` executions which return only segments part of the true condition.
+  
+* Lastly, the algorithm terminates only once the additional conditions check finds no additional conditions,
+  guaranteeing we have found the complete superset disjunctive condition.
+  
+## Making it More Performant
+
+As described above this approach can be slow since it processes glyphs one at a time. Improvements to performance
+can be made by processing glyphs in a batch. This can be done with a recursive approach where after each segment test 
+the set of input glyphs gets split into those that require the tested segment and those that don't. Each of the splits spawns
```

Comment:
There are also potential compromises to be made here on increasing the efficiency of both the algorithm and the patch table by decreasing the specificity. If you determine that segment `A` is relevant to glyph `x` and that segment `B` is relevant to glyph `y`, one can treat A, B as relevant to both. One might regret that depending on what happens next (how much relevance one combines by going down the road), but you can reduce glyph-specific processing and glyph-specific patch table entries in this way.